storage/innobase/buf/buf0flu.cc

```diff
@@ -140,6 +140,35 @@ inline void buf_pool_t::page_cleaner_wakeup()
     page_cleaner_is_idle= false;
     pthread_cond_signal(&do_flush_list);
   }
+  else if (pct_lwm != 0.0 && dirty_pct != 0)
```

Comment:
By definition, `dirty_pct` is always nonzero here. We do not need that condition. Because this function will be called while holding `buf_pool.flush_list_mutex` and right after adding something to `buf_pool.flush_list`, there will be some work to do, and we only need the condition
```c++
else if (pct_lwm != 0.0)
```